client/components/SearchBar.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import { fetchProducts } from '../store/products';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons';
+import Button from 'react-bootstrap/Button';
+
+function SearchBar(props, { placeholder }) {
+  let [filteredData, setFilteredData] = useState([]);
+  let [wordEntered, setWordEntered] = useState('');
+  let [products, setProducts] = useState(props.allProducts);
+
+  useEffect(() => {
+    props.getProducts();
+    setProducts(props.products);
+  }, []);
+
+  const handleFilter = (e) => {
+    const searchWord = e.target.value;
+    setWordEntered(searchWord);
+    const newFilter = props.products.filter((product) => {
```

Comment:
However expanding on above, when we're doing a filter we should never set an entire array of filtered data in local state. What we want to do instead is simply set a value to filter by in state, and filter the list we have from props in the render itself. Doing this will get rid of the broken setProducts pattern above anyway